src/vs/workbench/browser/parts/editor/editorGroupView.ts

```diff
@@ -286,7 +286,8 @@ export class EditorGroupView extends Themable implements IEditorGroupView {
 			if (this.isEmpty) {
 				EventHelper.stop(e);
 
-				this.openEditor(this.editorService.createEditorInput({ forceUntitled: true }), EditorOptions.create({ pinned: true }));
+				// {{SQL CARBON EDIT}} - use editor service to open editor, which will go through the override step and resolve to UntitledQueryEditorInput.
```

Comment:
What about setting the mode to 'sql' in the createEditorInput param? I just would rather we go through the same logic here in case they have a reason for doing it this way instead of using the editor service.

Reply:
here the command handler for "New Query" command: https://github.com/microsoft/azuredatastudio/blob/main/src/vs/workbench/contrib/files/browser/fileCommands.ts#L687

I'd prefer to use the same logic as this one.

Reply:
using 'sql' mode will open a normal file editor and set the language to SQL, it is different to the query editor:
![image](https://user-images.githubusercontent.com/13777222/123045119-60b2cc00-d3af-11eb-8a47-49521b980aca.png)

Reply:
Too bad, wonder why they're going through the logic here for opening the editor. Anyways not a huge deal I think given the small impact of the change.